### PR TITLE
Setup project

### PR DESCRIPTION
**Pull request by ATursunbekov**

### CODE REVIEW

**Comment by iPakTulane** [comment]
Subject to Siuzanna approval, everything looks good to be merged.

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Coordinator/BaseCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
+//
+//  BaseCoordinator.swift
+//  iOS_client_NeoCafe_
+//
+//  Created by Alikhan Tursunbekov on 4/2/24.
+//
+
+import Foundation
+import UIKit
+
```
Comment (iPakTulane) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Coordinator/BaseCoordinator.swift:10]: This is just to comment, for testing purposes.

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthView.swift [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        return image
+    }()
+    
+    lazy var customSegmentedControl: CustomSegmentedControl = {
+        let controller = CustomSegmentedControl()
+        controller.layer.cornerRadius = 24
+        return controller
+    }()
+    
+    lazy var textField = CustomTextField(placeholder: "Введите почту", image: Asset.dog.name)
```
Comment (siuzanna) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthView.swift:49]: swiftgen

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthView.swift [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    
+    lazy var button: UIButton = {
+        let button = UIButton()
+        button.setTitle("Получить код", for: .normal)
```
Comment (siuzanna) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthView.swift:53]: swiftgen

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthView.swift [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        }
+        
+        firstErrorText.snp.makeConstraints { make in
+            make.leading.trailing.equalToSuperview().inset(22)
+            make.top.equalTo(customSegmentedControl.snp.bottom).offset(40)
+        }
+        firstErrorText.isHidden = true
+    }
+    
+    required init?(coder: NSCoder) {
```
Comment (siuzanna) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthView.swift:153]: remove

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+//  Created by Alikhan Tursunbekov on 4/2/24.
+//
+
+import UIKit
+import SnapKit
+import SwiftUI
+
+class AuthViewController: UIViewController {
+    
+    let viewModel: AuthViewModelProtocol
```
Comment (siuzanna) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift:14]: private

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    let authView = AuthView()
```
Comment (siuzanna) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift:15]: lazy var and private

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    func cleanErrors() {
+        UIView.animate(withDuration: 0.5) {
+            self.authView.textField.snp.updateConstraints { make in
+                make.top.equalTo(self.authView.customSegmentedControl.snp.bottom).offset(40)
+            }
+            self.authView.textField.logo.tintColor = Asset.colorDarkBlue.color
+            self.authView.textField.textField.attributedPlaceholder = NSAttributedString(string: "Введите почту", attributes:
+                                                                                        [
+                                                                                            .foregroundColor: Asset.colorDarkGray.color,
+                                                                                            .font: UIFont(name: FontFamily.Poppins.regular.name, size: 16) ?? UIFont.systemFont(ofSize: 16)
```
Comment (siuzanna) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift:56]: extension for UIFont

.poppins(weight: .regular, size: 16)

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+                                                                            [
+                                                                                .foregroundColor: Asset.colorRed.color,
+                                                                                .font: UIFont(name: FontFamily.Poppins.regular.name, size: 16) ?? UIFont.systemFont(ofSize: 16)
+                                                                            ]
+            )
+        }
+    }
+    
+    @objc func textFieldChanged(_ sender: UITextField) {
+        if sender.text != "" {
```
Comment (siuzanna) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift:83]: sender.text != nil

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+            cleanErrors()
+        }
+    }
+    
+    @objc func nextPressed() {
+        if authView.textField.textField.text == "" {
```
Comment (siuzanna) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift:89]: nil

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+extension AuthViewController: SegmentedControlDelegate {
+    func valueChanged(selectedIndex: Int) {
+        if selectedIndex == 0 {
```
Comment (siuzanna) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift:103]: selectedIndex == 0 ?

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+            authView.titleLabel.text = "Вход"
+        } else {
+            authView.titleLabel.text = "Регистрация"
+        }
+        authView.textField.textField.text = ""
```
Comment (siuzanna) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/Authentication/AuthViewController.swift:108]: nil

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/CustomElements/CustomSegmentedControl.swift [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+        stackView.addArrangedSubview(firstButton)
+        stackView.addArrangedSubview(secondButton)
+        
+        stackView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+    
+    func selectionChanges(isFisrt: Bool) {
+        if isFisrt {
```
Comment (siuzanna) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/CustomElements/CustomSegmentedControl.swift:61]: ? :

File: iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/CustomElements/CustomSegmentedControl.swift [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+    }
+    
+    func updateSelectedState() {
+        switch selectedIndex {
+        case 0:
+            selectionChanges(isFisrt: true)
+        case 1:
+            selectionChanges(isFisrt: false)
+        default:
+            break
```
Comment (siuzanna) [inline iOS_client_NeoCafe_/iOS_client_NeoCafe_/Core/CustomElements/CustomSegmentedControl.swift:86]: Ternary operator

**Review by siuzanna — COMMENTED** [review]
remove extenstions

**Pull request merged**